beater/pub.go

```diff
@@ -16,16 +18,16 @@ import (
 // number requests(events) active in the system can exceed the queue size. Only
 // the number of concurrent HTTP requests trying to publish at the same time is limited.
 type publisher struct {
-	events  chan []beat.Event
-	client  beat.Client
-	m       sync.RWMutex
-	stopped bool
+	payloads chan pr.Payload
```

Comment:
Having the channel be a `pr.Payload` instead of `*pr.Payload` means we'll be copying all the data. Have you thought about making this channel of pointers instead? It might not make any difference.

Reply:
This shouldn't make a big difference, as the Payload mostly consists of Pointers and Slices.